### ISSUE TITLE
Clarify development instructions regarding llvmlite dev releases

### ISSUE THREAD

**Issue by esc** [numba/numba#7354]
As per Gitter:

<img width="761" alt="Screen Shot 2021-08-30 at 10 56 13" src="https://user-images.githubusercontent.com/41563/131313972-4ca1ad9e-259e-4d4e-aab1-e202ef3806a2.png">

We need to update the instructions at:

https://numba.readthedocs.io/en/latest/developer/contributing.html#build-environment?

So as to include installing an llvmlite release from. `numba/label/dev`.

**Comment by sgbaird** [comment]
This got rid of the error for me. Is this correct? ([Source](https://anaconda.org/numba/llvmlite))
`conda install -c numba/label/ci_llvmlite_0.38 llvmlite`

**Comment by sgbaird** [comment]
Also, to improve the searchability of this issue, here is the text version of the error:
```bash
ImportError: Numba requires at least version 0.38.0 of llvmlite.
Installed version is 0.37.0.
Please update llvmlite.
```

**Comment by esc** [comment]
@sgbaird thank you for looking into this. I am not sure where the `ci_llvmlite_0.38` channel came from. IIRC development releases of llvmlite can be obtained from the `numba/label/dev` channel on anaconda.org.

**Comment by esc** [comment]
the label on anaconda.org: `ci_llvmlite_0.38` was removed, it was a temporary storage for some experimental packages.

**Comment by FirefoxMetzger** [comment]
I ran into the same issue while setting up a dev environment (new to numba, and preparing my first PR 😄). I guess the correct instruction to fix this is to use the dev channel as suggested by @esc ?

```bash
conda install -c numba/label/dev llvmlite
```

**Comment by esc** [comment]
> I ran into the same issue while setting up a dev environment (new to numba, and preparing my first PR 😄). I guess the correct instruction to fix this is to use the dev channel as suggested by @esc ?
> 
> ```shell
> conda install -c numba/label/dev llvmlite
> ```

Yes, that is one correct way to obtain developmental releases of llvmlite.
